libbeat/processors/dns/config.go

```diff
@@ -0,0 +1,129 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package dns
+
+import (
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Config defines the configuration options for the DNS processor.
+type Config struct {
```

Comment:
This is a very complete config structure, but its quite clunky to write. 

```yaml
processors:
- dns:
    lookup:
      - type: reverse
        fields:
          message: resolved
```

This could be improved IMHO. Maybe to:

```yaml
dns:
  type: reverse
  fields:
    message: resolved
```

If you need multiple types of lookup you'd use different processors. An added bonus here is that you'd separate caches by type (forward vs. reverse).